File: java/resistor-color/src/main/java/ResistorColor.java
class ResistorColor {
    private final String[] resistorColors = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
    int colorCode(String color) {
        for (int i = 0; i < resistorColors.length; i++) {
            if (resistorColors[i].equals(color)) {
                return i;
            }
        }
        return -1;
    }

    String[] colors() {
        return this.resistorColors;
    }
}

File: c/resistor-color/src/resistor_color.c
#include "resistor_color.h"

static resistor_band_t COLORS[10] = {
  BLACK,
  BROWN,
  RED,
  ORANGE,
  YELLOW,
  GREEN,
  BLUE,
  VIOLET,
  GREY,
  WHITE,
};

int color_code(resistor_band_t color)
{
  return (int)color;
}

const resistor_band_t *colors()
{
  return COLORS;
}
